src/expr/macro/src/gen.rs

```diff
@@ -1223,17 +1255,19 @@ impl FunctionAttr {
                             #(let #inputs = unsafe { #arrays.value_at_unchecked(i) };)*
                             for output in #iter {
                                 index_builder.append(Some(i as i32));
-                                match #output {
-                                    Some((#(#outputs),*)) => { #(#builders.append(#optioned_outputs);)* }
-                                    None => { #(#builders.append_null();)* }
-                                }
+                                #append_output
 
                                 if index_builder.len() == self.chunk_size {
                                     let len = index_builder.len();
                                     let index_array = std::mem::replace(&mut index_builder, I32ArrayBuilder::new(self.chunk_size)).finish().into_ref();
                                     let value_arrays = [#(std::mem::replace(&mut #builders, #builder_types::with_type(self.chunk_size, #return_types)).finish().into_ref()),*];
                                     #build_value_array
-                                    yield DataChunk::new(vec![index_array, value_array], self.chunk_size);
+                                    let error_array = std::mem::replace(&mut error_builder, Utf8ArrayBuilder::new(self.chunk_size)).finish().into_ref();
```

Comment:
I think there might be no need to reserve capacity for the error message array builder. Also the implementation of reservation for `Utf8ArrayBuilder` makes little sense. 🤣

https://github.com/risingwavelabs/risingwave/blob/d19d8f9ae55832430f452c2ed90fca44910dd37b/src/common/src/array/bytes_array.rs#L142-L146

Reply:
Actually we should distinguish `item_capacity` and `data_capacity` like [arrow](https://docs.rs/arrow/latest/arrow/array/builder/struct.GenericByteBuilder.html#tymethod.with_capacity) does. If only one capacity can be specified, it should refer to `item_capacity`. And here we expect `item_capacity = chunk_size` and `data_capacity = 0`.

Reply:
fixed in 2c52ee3

Reply:
You're right. I just realized that even there's no error in the result, we still need to allocate the `offset` vector.

Reply:
Ideally, if there is no error, we can avoid building the error array at all. But that would make the code much more complicated.